pkg/systemd/overview-cards/usageCard.jsx

```diff
@@ -29,59 +29,113 @@ import "./usageCard.less";
+            this.metrics_channel.removeEventListener("message", this.onMetricsUpdate);
+            this.metrics_channel.close();
+            this.metrics_channel = null;
+            return;
+        }
+
+        if (!cockpit.hidden && this.metrics_channel === null) {
+            this.metrics_channel = cockpit.channel(METRICS_SPEC);
+            this.metrics_channel.addEventListener("closed", (ev, error) => console.error("metrics closed:", error));
+            this.metrics_channel.addEventListener("message", this.onMetricsUpdate);
+        }
     }
 
-    updateMemoryInfo() {
-        if (!this.state.pollingEnabled)
+    onMetricsUpdate(event, message) {
+        const data = JSON.parse(message);
+
+        // reset state on meta messages
+        if (!Array.isArray(data)) {
+            this.samples = [];
             return;
+        }
 
-        machine_info.cpu_ram_info().done(info => {
-            this.setState({
-                memTotal: Number((info.memory / (1024 * 1024 * 1024)).toFixed(1)),
-                memAvailable: Number((info.available_memory / (1024 * 1024 * 1024)).toFixed(1))
+        // decompress
+        data.forEach(samples => {
+            samples.forEach((sample, i) => {
+                if (sample !== null)
+                    this.samples[i] = sample;
             });
         });
-        window.setTimeout(this.updateMemoryInfo.bind(this), UPDATE_DELAY);
+
+        // CPU metrics are in ms/s; divide by 10 to get percentage
+        if (this.samples[0] !== false) {
+            const cpu = Math.round((this.samples[0] + this.samples[1] + this.samples[2]) / 10 / this.state.numCpu);
+            this.setState({ cpuUsed: cpu });
+        }
+        this.setState({ memUsed: Number((this.samples[3] / (1024 * 1024 * 1024)).toFixed(1)) });
     }
 
     render() {
-        const memUsed = Number((this.state.memTotal - this.state.memAvailable).toFixed(1));
-        const fraction = memUsed / this.state.memTotal;
+        const fraction = this.state.memUsed / this.state.memTotal;
+        const cores_str = cockpit.format(cockpit.ngettext("of $0 CPU core", "of $0 CPU cores", this.state.numCpu), this.state.numCpu);
 
         return (
             <Card className="system-usage">
                 <CardHeader>{_("Usage")}</CardHeader>
                 <CardBody>
                     <table className="pf-c-table pf-m-grid-md pf-m-compact">
                         <tbody>
+                            <tr>
+                                <th scope="row">{_("CPU")}</th>
+                                <td>
+                                    <Progress value={this.state.cpuUsed}
+                                        className="pf-m-sm"
+                                        min={0} max={100}
+                                        variant={ this.state.cpuUsed > 90 ? ProgressVariant.danger : ProgressVariant.info }
+                                        label={ this.state.cpuUsed + '% ' + cores_str }
```

Comment:
When the label changes width (which it does a lot since digits all seem to have different width in our font), the progress  bar also changes size.  That doesn't look nice.

Reply:
![image](https://user-images.githubusercontent.com/3228183/71976453-09528780-321f-11ea-83d6-3243532d5494.png)

![image](https://user-images.githubusercontent.com/3228183/71976490-1f604800-321f-11ea-98e9-6c9bc76ac9b8.png)

Reply:
@mvollmer: I can honestly not see the problem on your screenshots, but with a window width of just the right size, the progress bars do get very narrow to make room for the labels:

![overview-cpu-narrow](https://user-images.githubusercontent.com/200109/71976722-4d418e80-3217-11ea-8f11-8fe047881919.png)

Not sure how else to represent this, though. Would be nice to make the other progress bar the same length, of course.

Reply:
Yep, this is known. I think I have a work-around to minimize the issue, but it still exists. I wish we had a more complete font to be able to turn on fixed-width numbers (like open sans has), but that font feature doesn't exist in either Overpass or RedHatText (yet).

Reply:
There's another problem that the cards are getting too small for some reason. (The overview was modified when I was on vacation; it may be due to this. Or perhaps it was a last minute change I made? Anyway, regardless of the cause, I'm currently working on sizing optimizations and will take progress bar size into consideration too.)